### ISSUE TITLE
Does not work when you have a source build and a project

### ISSUE THREAD

**Issue by ughstudios** [boocs/ue4-intellisense-fixes#11]
So if you are using unreal game sync, you are required to have an Engine and a path together
![image](https://user-images.githubusercontent.com/1284431/156867972-7d52e6a8-fc20-45d2-b148-c5d6d00952e8.png)
You can see in the image. If I setup vscode i get the squiggles. I tried this extension and it didn't do anything.

**Comment by ughstudios** [comment]
```
Searching for *.uproject file...
No *.uproject file found!
```

**Comment by ughstudios** [comment]
If I create a symlink I get the following error:
```
Searching for *.uproject file...
Extension "UE Intellisense Fixes" is now active!

Found Unreal Engine v4.27.1

** Error **: This fixable project hasn't been initialized properly and is invalid.

```

**Comment by boocs** [comment]
Sorry! This is the first I've heard of UGS so this could take awhile. 

So UGH1 is a project inside the UE folder?

Can you show a picture of everything that's in the UGH1 folder?

**Comment by ughstudios** [comment]
![image](https://user-images.githubusercontent.com/1284431/156869863-229da96c-c896-4cd9-bf4e-a4cf83bf2471.png)
The project is in UGH1/UGH1, I created a symlink to the file in `C:\repos\Perforce\daniel_main\UGH1\UGH1\UGH1.uproject`

UGH1 is the root of my perforce workspace which has the Engine source in `Engine/` and then `UGH1/UGH1` is the actual game project. Does that make sense?

**Comment by boocs** [comment]
Yep makes sense!

What about a pic of what's in UGH1/UGH1?

**Comment by ughstudios** [comment]
![image](https://user-images.githubusercontent.com/1284431/156871360-992f2bca-13d4-481c-8be7-147c54a60d1c.png)
Just a normal unreal project.

**Comment by ughstudios** [comment]
I should mention that the sym link does NOT fix the problem at all, so I'm deleting it.

**Comment by boocs** [comment]
I think you need to create a vscode project for the UGH1/UGH1 project. 

The first UGH1 folder has a vscode workspace file inside but that's a workspace project for modifying the unreal engine source code not for modifying a individual project.

In the script or scripts folder of UGH1/UGH1 is there a script for creating a vscode workspace project for UGH1/UGH1?

**Comment by boocs** [comment]
If not, you might be able to right click on the UGH1/UGH1 UGH1.uproject file and select the option Generate Visual Studio project files....

Is it there? What happens if you choose it?

**Comment by ughstudios** [comment]
![image](https://user-images.githubusercontent.com/1284431/156893967-00ff90ce-4c0f-434d-bb84-416f04b78c7e.png)
There's nothing there. Also, if you try to generate project files, nothing happens.

**Comment by ughstudios** [comment]
![image](https://user-images.githubusercontent.com/1284431/156893999-cdd92b1d-7601-4560-be3e-122e6c9a185b.png)
Also it looks like it creates the UGH1 folder here for the project in the .vscode directory

**Comment by boocs** [comment]
That compile commands folder should be for the first UGH1 folder not the second UGH1 project folder.

Can you double click the UGH1.uproject file, to open the project in the Unreal Engine, and then switch the engine to use VSCode?

It's in Editor Preferences -> Source Code -> Source Code Editor : Switch it to VSCode

**Comment by ughstudios** [comment]
Yeah I did that

**Comment by boocs** [comment]
Now again try double clicking the UGH1/UGH1 UGH1.uproject file, to load the project into the UE, and in the File menu there should be something that says Refresh Visual Studio Code Project. I believe it in another menu for UE5 but in UE4 it should be there.

**Comment by boocs** [comment]
Another thing, are there any C++ files in the UGH1/UGH1 Source folder/subfolders? If not then UE might think it's a Blueprint project and won't setup the project with VSCode/C++ config files.

**Comment by ughstudios** [comment]
Yes there are cpp files and I also clicked refresh multiple times.

**Comment by boocs** [comment]
I guess it could be some weird/different UGS config thing. 

If you open the UE4.code-workspace in a text editor can you post what's inside. Just to see if it even mentions the UGH1/UGH1 folder.

**Comment by ughstudios** [comment]
```
{
	"folders": [
		{
			"name": "UE4",
			"path": "."
		}
	],
	"settings": {
		"typescript.tsc.autoDetect": "off"
	},
	"extensions": {
		"recommendations": [
			"ms-vscode.cpptools",
			"ms-dotnettools.csharp"
		]
	}
}

```

**Comment by ughstudios** [comment]
![image](https://user-images.githubusercontent.com/1284431/156944652-f794a89b-e7af-4536-9a8c-84af64ffb11a.png)

**Comment by boocs** [comment]
Check out this post from NicholasMont to generate project file from the command line. Maybe it'll give you some message that helps with not creating project files for UGH1/UGH1.

https://forums.unrealengine.com/t/why-cant-i-generate-visual-studio-project-files/434992/6

**Comment by ughstudios** [comment]
![image](https://user-images.githubusercontent.com/1284431/157094624-4220386f-e992-4add-9c2c-297f81279671.png)
Doesn't seem like it did anything

**Comment by boocs** [comment]
I believe this is the same thing you're doing: https://forums.unrealengine.com/t/whats-a-native-project/264986

UE4-Main is your first UGH1 folder and VanillaProj is your UGH1/UGH1 folder. In the VanillaProj it has a .vs and a VanillaProj.sln. 

So I'm guessing I'm correct in saying that they should be there. In your case a .vscode and .code-workspace file.

You'll probably have better luck asking in another forum on how to generate project files for native projects.

My last suggestion would be deleting(or moving just in case) these folders in  UGH1/UGH1 and trying the same thing with creating project file on the command line. You can skip the part you already did with setting the source code setting to VSCode in UE.

https://forums.unrealengine.com/t/generate-visual-studio-project-files-not-working/417083
